app/src/userpages/components/StreamPage/Show/SecurityView/index.jsx

```diff
@@ -0,0 +1,183 @@
+// @flow
+
+import React, { useCallback } from 'react'
+import { connect } from 'react-redux'
+import { I18n } from 'react-redux-i18n'
+import cx from 'classnames'
+
+import type { Stream } from '$shared/flowtype/stream-types'
+import type { StoreState } from '$shared/flowtype/store-state'
+import { updateEditStream } from '$userpages/modules/userPageStreams/actions'
+import { selectEditedStream } from '$userpages/modules/userPageStreams/selectors'
+import Slider from './Slider'
+import * as Icons from './Icons'
+
+import styles from './securityView.pcss'
+
+type OwnProps = {
+    disabled: boolean,
+}
+
+type StateProps = {
+    stream: ?Stream,
+}
+
+type DispatchProps = {
+    updateEditStream: (Stream) => void,
+}
+
+type Props = OwnProps & StateProps & DispatchProps
+
+/*
+ * The security slider manipulates two boolean fields on the Stream object.
+ * The slider setting maps to these booleans as follows:
+ * Basic:
+ * requireSignedData: false
+ * requireEncryptedData: false
```

Comment:
Don't know if I have an older e&e instance running but `requireEncryptedData` is not saved or returned but the stream endpoint. If you set the selection to "Encrypted", it does not remember the choice after refreshing the page.